File: common/src/main/java/cn/breadnicecat/candycraftce/misc/CSugarFuels.java
package cn.breadnicecat.candycraftce.misc;

import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import org.jetbrains.annotations.NotNull;

import static cn.breadnicecat.candycraftce.item.CItems.PEZ;
import static cn.breadnicecat.candycraftce.utils.CommonUtils.apply;

public class CSugarFuels {
	public static final Object2IntOpenHashMap<Item> FUELS = apply(new Object2IntOpenHashMap<>(), (m) -> {
		m.put(PEZ.get(), 1000);
	});

	public static boolean isFuel(Item item) {
		return item != Items.AIR && FUELS.containsKey(item);
	}

	public static boolean isFuel(@NotNull ItemStack stack) {
		return !stack.isEmpty() && FUELS.containsKey(stack.getItem());
	}

	public static int getBurnDuration(Item item) {
		return item == Items.AIR ? 0 : FUELS.getInt(item);
	}

	public static int getBurnDuration(@NotNull ItemStack stack) {
		return stack.isEmpty() ? 0 : FUELS.getInt(stack.getItem());
	}
}

File: src/main/kotlin/cn/breadnicecat/candycraftce/core/misc/CSugarFuels.kt
package cn.breadnicecat.candycraftce.core.misc

import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items.AIR

object CSugarFuels {

    fun isFuel(item: Item): Boolean {
        return item !== AIR //todo
    }

    fun isFuel(stack: ItemStack): Boolean {
        return isFuel(stack.item)
    }

    fun getBurnDuration(item: Item?): Int {
        return if (item === AIR) 0 else 1//todo
    }

    fun getBurnDuration(stack: ItemStack): Int {
        return if (stack.isEmpty) 0 else 1 //todo
    }
}
